src/RepositoryAPI.jl

```diff
@@ -456,9 +137,151 @@ end
+            object = getRepositoryDomain(token, appName = name)
+            objectID = get(object, "id", 0)
+        elseif objectType == "tenant"
+            object = getRepositoryTenant(token, name = name)
+            objectID = get(object, "id", 0)
+        end
+    end
+
+    local isKeySet = false
+
+    if filterRequired
+        isKeySet = any(kv -> isa(kv[2], Number) ? kv[2] > 0 : !isempty(kv[2]), searchKey)
+
+        if !isKeySet
+            throw(ArgumentError("At least one of `$(join(collect(keys(searchKey)), "', `", "' or `"))' must be set"))
+        end
+    end
+
+    local url = ObjectEndpoint * "/" * objectType * "/$(objectID)"
+    local debugID = "(all)"
+
+    if verbose
+        println("POST $url")
+        println("X-Auth-Token: $token")
+        println(attributes)
+    end
+
+
+    json = Dict{AbstractString, Any}()
+    json["type"] = objectType
+
+    # If attributes is supplied, update the object’s attributes field
+    if !isempty(attributes)
+        attributesDict = []
+
+        for (key, val) in attributes
+            push!(attributesDict, Dict("name" => key, "value"=> val))
+        end
+        
+        json["attributes"] = attributesDict
+    end
+
+    # If any objectFields are supplied by the user, update these in the object (if it exists)
+    if !isempty(objectFields)
+        for (key, val) in objectFields
+            json[key] = val
+        end
+    end
+
+    resp = Requests.post(url,
+        json = json,
+        headers = Dict("X-Auth-Token" => token, "Content-type" => "application/json")
+    )
+
+    if statuscode(resp) != 204
+        error("Error updating $(objectType), id = $(objectID).")
+    end
+
+    return resp
+
+end
+
+
+
+# Internal convenience function.  Deletes an object from the repository.
+function deleteRepositoryObject(token::AbstractString,
+                              objectType::AbstractString,
+                              searchKey::Dict{Symbol, Any}
+)
+
+    global verbose
+
+    if token == ""
+        throw(ArgumentError("`token' cannot be empty"))
+    end
+
+    objectID = get(searchKey, :id, 0)
+    name = get(searchKey, :name, "")
+
+    # If objectID is not supplied, retrieve from get function
+    if objectID == 0 
+        if objectType == "alert"
+            object = getRepositoryAlert(token, alertName = name)
+            objectID = get(object, "id", 0)
+        elseif objectType == "domain"
+            object = getRepositoryDomain(token, appName = name)
+            objectID = get(object, "id", 0)
+        elseif objectType == "tenant"
+            object = getRepositoryTenant(token, name = name)
+            objectID = get(object, "id", 0)
+        end
+    end
+
+    local url = ObjectEndpoint * "/" * objectType * "/$(objectID)"
+    local debugID = "(all)"
+
+    if verbose
+        println("DELETE $url")
+        println("X-Auth-Token: $token")
+    end
+
+    resp = Requests.delete(url,
+        json = json,
```

Comment:
where did `json` come from?

Reply:
That's a mistake, shouldn't be there.  Removed